docs/core-concepts/bot-sequences/actions.mdx

```diff
@@ -1,9 +1,624 @@
+            }
+        ],
+        "keyboard": [ // Any number of keyboard inputs can be provided
+            {
+                "startTime": ..., // A number indicating the time offset at which the action should be performed from the start of the segment (in seconds)
+                "endTime": ..., // A number indicating the time offset at which the action should be stopped from the start of the segment (in seconds)
+                "binding": ..., // A sting representing the binding to be pressed
+            }
+        ]
+    }
+}
+```
+
+**Notes**
+
+- You can use the `startTime` and `endTime` fields to control the timing of actions. For instance, you can hold shift and press a key to result in a capital letter.
+- If two actions have the same `startTime`, the action that appears earlier in the list will be performed first.
+- You can omit an `endTime` of a keyboard action to have the action continue until it is disabled using using an action that only sets the `endTime`.
+- The available keys to press in keyboard actions can be found [here](https://github.com/Regression-Games/RGUnityBots/blob/cd68ed0f4b10567eb652477717338d92b4f4571e/src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/KeyboardInputActionObserver.cs#L25-L151). The bindings should take the format of `<Keyboard>/keyname`.
+
+**Example: Typing "Hello"**
+
+```json
+{
+    "name": "Type 'Hello'",
+    "description": "Types 'Hello' using the keyboard",
+    "endCriteria": [...],
+    "botAction": {
+        "type": "InputPlayback",
+        "data": {
+            "startTime":0.0,
+            "inputData": {
+                "keyboard": [
+
+                    {"startTime":0.0,"binding":"<Keyboard>/leftShift"},
+                    {"startTime":0.0,"endTime":0.0,"binding":"<Keyboard>/h"},
+                    {"endTime":0.0,"binding":"<Keyboard>/leftShift"},
+
+                    {"startTime":0.0,"endTime":0.0,"binding":"<Keyboard>/e"},
+                    {"startTime":0.0,"endTime":0.0,"binding":"<Keyboard>/l"},
+                    {"startTime":0.0,"endTime":0.0,"binding":"<Keyboard>/l"},
+                    {"startTime":0.0,"endTime":0.0,"binding":"<Keyboard>/o"}
+
+                ],
+                "mouse":[]
+            }
+        }
+    }
+}
+```
+
+**Example: Moving and clicking the mouse**
+
+```json
+{
+    "name": "Move and click the mouse",
+    "description": "Moves the mouse to a position and clicks it",
+    "endCriteria": [...],
+    "botAction": {
+        "type": "InputPlayback",
+        "data": {
+            "startTime": 0.0,
+            "inputData": {
+                "keyboard": [],
+                "mouse": [
+                    {"startTime":0.0,"screenSize":{"x":1920,"y":1080},"position":{"x":942,"y":500},"worldPosition":null,"leftButton":false,"middleButton":false,"rightButton":false,"forwardButton":false,"backButton":false,"scroll":{"x":0,"y":0},"clickedObjectNormalizedPaths":["UI Canvas/Start Button"]},
```

Comment:
it is good that this is here, we should add a comment stating that it is best practice to have a mouse position input before the click input .  Without this, Unity's input handling can cause some un-expected behaviour with the click especially when the prior mouse position was over a different input component in the UI.